src/main/java/com/ust/sourcecourse/configuration/service/HomeService.java

```diff
@@ -7,52 +7,83 @@
+			response.setTables(project.getProjectTables().size());
+			response.setGroups(project.getGroups().size());
+			homePageResponses.add(response);
 		}
+		return homePageResponses;
+	}
+
+	public List<String> getDescriptionAndTags(Long uid) {
+		SourceTable homePage = sourceTableRepository.findById(uid).orElseThrow();
+
+		return homePage.getTags();
+	}
+
+	public DBTableColumn UpdateDescriptionandTags(Long uid, HomeRequest request) {
+		SourceColumn homePage = sourceColumnRepository.findById(uid).orElseThrow();
+		homePage.setDescription(request.getDescription());
+		homePage.setTags(request.getTags());
+		SourceColumn SchemaPage = sourceColumnRepository.save(homePage);
+		return getDBTableColumn(SchemaPage);
+	}
+
+	private DBTableColumn getDBTableColumn(SourceColumn sourceColumn) {
+		DBTableColumnMetadata metadata = DBTableColumnMetadata.builder().type(sourceColumn.getType())
+				.isPrimary(sourceColumn.isPrimary()).isNullable(sourceColumn.isNullable())
+				.isUnique(sourceColumn.isUnique()).defaultValue(sourceColumn.getDefaultValue()).build();
+		return DBTableColumn.builder().uid(sourceColumn.getUid()).name(sourceColumn.getName())
+				.description(sourceColumn.getDescription()).metadata(metadata).tags(sourceColumn.getTags()).build();
+	}
+
+	public DBTableColumn UpdateDescriptionAndRemoveAllTags(Long uid, HomeRequest request) {
+		SourceColumn sourceColumn = sourceColumnRepository.findById(uid).orElseThrow();
+		sourceColumn.setDescription(request.getDescription());
+		sourceColumn.getTags().clear(); // Remove all tags
+		SourceColumn updatedSourceColumn = sourceColumnRepository.save(sourceColumn);
+		return getDBTableColumn(updatedSourceColumn);
+	}
+
+	public void deleteTableTags(Long uid) {
+		SourceTable sourceTable = sourceTableRepository.findById(uid)
+				.orElseThrow(() -> new EntityNotFoundException("Table not found"));
+		sourceTable.getTags().clear();
+		sourceTableRepository.save(sourceTable);
+	}
 
-//		public Project createProject(Project project) {
-//			return projectRepository.save(project);
-//		}
-//		
-//		public Project getProjectById(Long id) {
-//			return projectRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Project not found with id: " + id));
-//		}
-//
-//		public List<Project> getAllProjects() {
-//			return projectRepository.findAll();
-//		}
-//		
-//		public Project updateProject(Long id, Project project) {
-//			Project existingProject = projectRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Project not found with id: " + id));
-//			existingProject.setName(project.getName());
-//			existingProject.setDescription(project.getDescription());
-//			existingProject.setModifiedBy(project.getModifiedBy());
-//			existingProject.setModifiedTimestamp(LocalDateTime.now());
-//			return projectRepository.save(existingProject);
-//		}
-//		
-//		public void deleteProject(Long id) {
-//			projectRepository.deleteById(id);
-//		}
+	public void deleteColumnTags(Long uid) {
```

Comment:
Similar to above, 
Why deleting all tags. API should be to remove list of tags. Not remove all tags. 
Request payload should have list of tags to be removed

Reply:
resolved